design-doc.md

```diff
@@ -0,0 +1,179 @@
+*   [https://docs.google.com/document/d/1smOrLXBs4E9KvRcjFzVFSGRLSOdJFioi8HHv_xvMlCI/edit](https://docs.google.com/document/d/1smOrLXBs4E9KvRcjFzVFSGRLSOdJFioi8HHv_xvMlCI/edit)
+*   [https://docs.google.com/document/d/1hFMPtjQnlscUwIauxcjAPxFmRAgruZ-6TBnKGEW4PWc/edit#heading=h.3svxzuabblb4](https://docs.google.com/document/d/1hFMPtjQnlscUwIauxcjAPxFmRAgruZ-6TBnKGEW4PWc/edit#heading=h.3svxzuabblb4)
+*   [https://docs.google.com/document/d/1-V07kO9L1mab8GV1-OQoLTAIDPZyWi5P_PRoIsEHeik/edit#heading=h.3svxzuabblb4](https://docs.google.com/document/d/1-V07kO9L1mab8GV1-OQoLTAIDPZyWi5P_PRoIsEHeik/edit#heading=h.3svxzuabblb4)
+*   [https://docs.google.com/document/d/1fiRlWzmeHYXGtMXEUCcs82sLQDyZc1oSdY5lUs566jc/edit](https://docs.google.com/document/d/1fiRlWzmeHYXGtMXEUCcs82sLQDyZc1oSdY5lUs566jc/edit)
+*   [https://docs.google.com/document/d/1p8IW_QuzMO6EhLiBOPJY0FqD0ZoS64-7M_aZvP9y2-8/edit#](https://docs.google.com/document/d/1p8IW_QuzMO6EhLiBOPJY0FqD0ZoS64-7M_aZvP9y2-8/edit#)
+*   [https://github.com/dcos-labs/ui-kit/blob/master/CONTRIBUTING.md](https://github.com/dcos-labs/ui-kit/blob/master/CONTRIBUTING.md)
+*   [https://docs.google.com/document/d/1p-CnUnhNFVz6WomPQvMEsTybbGgV0FHWgsUzBqX64cs/edit#heading=h.apvnjfls8tkq](https://docs.google.com/document/d/1p-CnUnhNFVz6WomPQvMEsTybbGgV0FHWgsUzBqX64cs/edit#heading=h.apvnjfls8tkq)
+
+
+## Living Document
+
+This is a living document[^1]. Feel free to add or change information. Everybody should be able to contribute to this document.
+
+
+## General Guidelines
+
+
+### Why are we building the design system?
+
+Have one source of truth, build a consistent product across company
+
+and platforms, increase developer productivity and feature delivery.
+
+
+### When does a component go into ui-kit
+
+Ui-kit follows the rules of three, which means once a component is used in 3 different places it should be added to ui-kit.
+
+
+### Flexibility
+
+We are trying to avoid unnecessary flexibility of a component. There is always room for building on the existing component or creating a new component for a different use case. Always when adding flexibility for another use case consider adding a new component instead or ensure that the complexity of the component is not exploding.
+
+
+### Considerations
+
+
+#### Avoid weak elements
+
+Weak elements are elements which can’t nest block elements, for example, a `p` tag is not able to wrap around a div which will reduce the comparability of a component. This is why we try to rather use a strong element like `section` or `div` instead of `p`.
+
+
+#### Declare Prop Types
+
+Since the `ui-kit` is using typescript please make sure to define proper prop types for your components. This will improve the documentation, which creates dynamically and will provide better error messages for the users. And also this will provide auto-completion in some IDE’s. Describe the props as well as possible, so also define object shapes or array shapes.
```

Comment:
Suggestions:
"which creates dynamically" -> "which **is created** dynamically"
"And also this will provide" -> "And also **these types** will provide"
"Describe the props as well as possible" -> "Describe the props as **good** as possible"
"so also define object shapes or array shapes." -> "**including** object **types** **and** array **types**."